src/Microsoft.AspNet.Http.Extensions/ResponseExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Http.Features;
+
+namespace Microsoft.AspNet.Http.Extensions
+{
+    public static class ResponseExtensions
+    {
+        public static bool CanClear(this HttpResponse response)
+        {
+            return !response.HasStarted && response.Body.CanSeek;
+        }
+
+        public static void Clear(this HttpResponse response)
+        {
+            response.StatusCode = 200;
```

Comment:
All of these APIs should already throw. The only reason to throw ourselves is if we want a more specific error message.